### ISSUE TITLE
FERR-74 - Restrict search/view by entity searching across multiple entities using a flag

### ISSUE THREAD

**Issue by nlaprell** [marklogic-community/data-explorer#44]
**Original Reporter:**  @tvnpraveen  
**Created:** 	28/Nov/17 11:52 AM
# Description
In our case lets say we have different entities with envelope patterns

Example:

1. StudySites
2. Audits
3. SubjectVisits

Once I create a query template with a pattern :envelope/:content/:StudySites on the StudySites entity, I expected to see only StudySites entities in the results view. 

We can have a checkbox in search view below the DocType which can be called Strict or Filter which restricts to that entity based on the xpath in DocType.

Please find the attachment where searching on StudySites is retrieving all entities.

# Attachments
![image](https://user-images.githubusercontent.com/22891553/35055770-2ef160fa-fb6d-11e7-8019-170cce4240bc.png)

**Comment by damonfeldman** [comment]
I suggest the behavior for this should be to restrict the query to the node type (e.g. StudySites) but allow query and view selection from all elements, including those in the header. A typical envelope will match the entity services spec and look like:

```
<es:envelope xmlns:es="http://marklogic.com/entity-services">
    <headers>
        <title>President</title>
   </headers>
   <es:instance>
      <Person>
        <id>1234</id>
        <firstName>George</firstName>
        <lastName>Washington</lastName>
      </Person>
    </es:instance>
</es:envelope>
```

and the 3rd level element <Person> is the one that distinguishes the type of entity, so we want to query only for docs containing <Person>.

But we do want to be able to query and display <title> because that's high-value data.

So this should restrict the set of docs, but no more than that.

**Comment by josvr** [comment]
OK I will implement this suggestion of Damon.

**Comment by damonfeldman** [comment]
Exactly.

BTW, I think the term "root" is now incorrect and confusing. On CMS the root node of the XML told you the document type quite accurately, so they chose a "root." But we really want a **type-discriminator** element, where if it exists we have a certain type of document. It does not need to be the root (that was just some fuzzy thinking at CMS 5 years ago).

**Comment by josvr** [comment]
Solved in the latest commit to #13 FERR-90 branch

**Comment by markschiffner** [comment]
Closed with FERR-90 merge
